frontend/public/components/storage/attach-storage.tsx

```diff
@@ -40,113 +27,136 @@ const PVCDropdown: React.SFC<PVCDropdownProps> = props => {
+  };
+
+  const getMountPaths = (podTemplate: any) => {
+    const tmpContainers = _.get(podTemplate, 'spec.containers', []);
+    return tmpContainers.reduce((acc, container) => {
+      if (!isContainerSelected(container)) {
+        return acc;
+      }
+      const mountPaths = _.map(container.volumeMounts, 'mountPath');
+      return acc.concat(mountPaths);
+    }, []);
   };
 
-  handleChange: React.ReactEventHandler<HTMLInputElement> = event => {
-    const { name, value } = event.currentTarget;
-    this.setState({ [name]: value } as any);
+  const checkMountPaths = (path: string) => {
+    const existingMountPaths = getMountPaths(resourceObj.spec.template);
+    const err =
+      existingMountPaths.indexOf(path) > -1
+        ? 'Mount path is already in use.'
+        : '';
+    setError(err);
   };
 
   // Add logic to check this handler for if a mount path is not unique
-  handleMountPathChange: React.ReactEventHandler<HTMLInputElement> = event => {
-    const { value: mountPath } = event.currentTarget;
-    this.setState({ mountPath });
+  const handleMountPathChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    setMountPath(event.currentTarget.value);
     // Look at the existing mount paths so that we can warn if the new value is not unique.
-    this.checkMountPaths(mountPath);
+    checkMountPaths(event.currentTarget.value);
+  };
+
+  const handleSubPathChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    setSubPath(event.currentTarget.value);
   };
 
-  handlePVCChange = (claimName: string) => {
-    this.updateVolumeName(claimName);
-    this.setState({ claimName });
+  const handlePVCChange = (newClaimName: string) => {
+    updateVolumeName(newClaimName);
+    setClaimName(newClaimName);
   };
 
-  onMountAsReadOnlyChanged: React.ReactEventHandler<HTMLInputElement> = () => {
-    this.setState({ mountAsReadOnly: !this.state.mountAsReadOnly });
+  const onMountAsReadOnlyChanged: React.ReactEventHandler<HTMLInputElement> = () => {
+    setMountAsReadOnly(!mountAsReadOnly);
   };
 
-  createPVCIfNecessary(): Promise<string> {
-    const { showCreatePVC, newPVCObj, claimName } = this.state;
+  const createPVCIfNecessary = ()=> {
     return showCreatePVC === 'new'
       ? k8sCreate(PersistentVolumeClaimModel, newPVCObj).then(claim => claim.metadata.name)
       : Promise.resolve(claimName);
-  }
+  };
 
-  save = (event: React.FormEvent<EventTarget>) => {
+  const save = (event: React.FormEvent<EventTarget>) => {
     event.preventDefault();
-    const { kindObj } = this.props;
-    const {
-      resourceObj: originalObj,
-      volumeName,
-      mountPath,
-      subPath,
-      mountAsReadOnly,
-      volumeAlreadyMounted,
-    } = this.state;
-    const { metadata } = originalObj;
-    const resourceObj = _.cloneDeep(originalObj);
-    const template = resourceObj.spec.template;
+    const updatedResourceObj = _.cloneDeep(resourceObj);
```

Comment:
Not new, but it would be better if we could use `k8sPatch`

Reply:
I will convert it to patch

Reply:
Converted this to a k8sPatch